pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -246,9 +246,13 @@ func (m *ManagerImpl) GetWatcherHandler() watcher.PluginHandler {
 }
 
 // ValidatePlugin validates a plugin if the version is correct and the name has the format of an extended resource
-func (m *ManagerImpl) ValidatePlugin(pluginName string, endpoint string, versions []string) error {
+func (m *ManagerImpl) ValidatePlugin(pluginName string, endpoint string, versions []string, foundInDeprecatedDir bool) error {
 	klog.V(2).Infof("Got Plugin %s at endpoint %s with versions %v", pluginName, endpoint, versions)
 
+	if foundInDeprecatedDir {
```

Comment:
> this completely separate from the storage driver code, and just implements the same PluginHandler interface?

Correct.

> is the same plugin watcher used to feed events to both CSI and device manager PluginHandler impls? 

Yes.

> if so, doesn't the plugin dir change in 1.13 affect device plugins as well?

Yes, it does. Per the discussions https://github.com/kubernetes/kubernetes/pull/70494#issuecomment-439509841 the owners of device plugin code @vishh @jiayingz are ok with the change.

> what stage are device plugins at, and how many releases do we need to honor them in the v1.12 location?

 @vishh @jiayingz can probably answer this better then I can.

Reply:
Device plugins haven't yet adopted the switch to using plugin watcher yet and we believe the risk very *very low* to make the switch to the new directory format.
However, given that this PR is supporting old plugin watcher implementation anyways for CSI, it might make sense to support for device plugins too just for the sake of consistency.

Reply:
Ok cool. I'll go ahead and remove the failure in this case then.

Reply:
> However, given that this PR is supporting old plugin watcher implementation anyways for CSI, it might make sense to support for device plugins too just for the sake of consistency.

I agree with that, thanks. Make sure the deprecation of the old dir is prominently announced for both device plugins and csi plugins in 1.13 release notes.